eq_cir_converter_service/services/schema/schema_processor_service.py

```diff
@@ -1,12 +1,53 @@
 """This module converts the schema from the current to the target version."""
 
+import copy
+from collections.abc import Mapping, Sequence
+
+from jsonpath_ng.ext import parse
+
 from eq_cir_converter_service.config.logging_config import logging
-from eq_cir_converter_service.types.custom_types import ConvertedSchema, InputSchema
+from eq_cir_converter_service.services.schema.extractable_strings import (
+    EXTRACTABLE_STRINGS,
+)
+from eq_cir_converter_service.types.custom_types import Schema
+from eq_cir_converter_service.utils.helper_utils import process_element
 
 logger = logging.getLogger(__name__)
 
 
-async def convert_schema(current_version: str, target_version: str, schema: InputSchema) -> ConvertedSchema:
+# --- JSONPath-Based Transformation ---
+def transform_json_schema(schema: Schema, paths: Sequence[Mapping[str, str]]) -> Schema:
+    """Transforms the JSON schema based on the provided JSONPath expressions.
+
+    Parameters:
+    - schema: The input schema to transform.
+    - paths: A sequence of dictionaries containing JSONPath expressions to apply.
+
+    Returns:
+    - A new schema with the transformations applied.
+    """
+    converted_schema = copy.deepcopy(schema)  # Work on a deep copy
+
+    for path_expr in paths:
+        expr = parse(path_expr["json_path"])
+        for match in expr.find(converted_schema):
+            context = match.context.value
+            key = match.path.fields[0] if hasattr(match.path, "fields") else None
+            index = getattr(match.path, "index", None)
+
+            if isinstance(context, dict) and key:
```

Comment:
I would take advantage of the walrus operator here (for both key and index) to improve readability and simplify it by initialising the index only when it is needed, specifically after the `isinstance()` check for a dictionary, rather than before.
Check my example from the previous comment.